firebase-firestore/src/main/java/com/google/firebase/firestore/core/Query.java

```diff
@@ -134,6 +134,9 @@ public boolean isCollectionGroupQuery() {
   /**
    * Returns true if this query does not specify any query constraints that could remove results.
    */
+  // TODO(ehsann): If we do allow composite filters with zero subfilters, then this should be
+  // updated.
+  // TODO(ehsann): How do we handle conditions that are always true? (e.g. age > 0 || age <= 0).
```

Comment:
We should not write any code to optimize this.

Reply:
Acknowledged.